pkg/retrier/retrier.go

```diff
@@ -48,6 +53,14 @@ func WithMaxRetries(maxRetries int, backOffPeriod time.Duration) RetrierOpt {
 	}
 }
 
+// WithNSecondsWaitRetries sets a retry policy that will retry forever with N-seconds wait time
+// between retries
+func WithNSecondsWaitRetries(waitSeconds int) RetrierOpt {
```

Comment:
Can we take here a `time.Duration` instead of int?
```suggestion
func WithBackOff(backOffPeriod time.Duration) RetrierOpt {
```